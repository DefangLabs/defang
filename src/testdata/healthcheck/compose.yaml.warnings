 ! can only have more than one replica for PRO tier and PRODUCTION deployment mode. Will set replicas to 1 for service(s) [cmd-shell curl flask1 flask2 none wget]
 ! service "cmd-shell": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "curl": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "flask1": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "flask2": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "none": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "wget": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors