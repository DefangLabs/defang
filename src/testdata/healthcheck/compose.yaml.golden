name: healthcheck
services:
  cmd-shell:
    healthcheck:
      test:
        - CMD-SHELL
        - echo hello
        - second line
    image: alpine
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5000
        protocol: tcp
  curl:
    healthcheck:
      test:
        - CMD
        - curl
        - -f
        - localhost
    image: curl
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
        protocol: tcp
  flask1:
    healthcheck:
      test:
        - CMD
        - python
        - -c
        - import sys, urllib.request;urllib.request.urlopen(sys.argv[1]).read()
        - http://localhost/
      timeout: 1s
    image: flask
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5000
        protocol: tcp
  flask2:
    healthcheck:
      test:
        - CMD
        - python
        - -c
        - import urllib.request;urllib.request.urlopen('http://127.0.0.1/path').read()
      interval: 10s
    image: flask
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5000
        protocol: tcp
  none:
    healthcheck:
      test:
        - NONE
        - ignored
      timeout: 1s
      interval: 1m0s
      retries: 3
    image: alpine
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5000
        protocol: tcp
  wget:
    healthcheck:
      test:
        - CMD
        - wget
        - -q
        - --spider
        - localhost:80
    image: alpine
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
        protocol: tcp
networks:
  default:
    name: healthcheck_default
